content/using-pytest-with-django.md

```diff
@@ -0,0 +1,176 @@
+
+# How?
+
+Let's dive into how you can setup `pytest` on your Django projects. For this post, I've created an repo with a dummy Django 1.8 [project](https://github.com/TrackMaven/using-pytest-with-django).  It's local development environment is managed by [Docker + Docker Compose.](http://engineroom.trackmaven.com/blog/a-better-development-environment-with-docker-and-fig/). All code examples are pulled from there.
+
+We'll need one more library in addition to `pytest`  to get everything working smoothly with Django. [`pytest-django`](https://pytest-django.readthedocs.org) takes care of replicating [Django's existing testing functionality](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#transactiontestcase) into `pytest`.
+
+In our [`requirements-dev.txt`](https://github.com/TrackMaven/using-pytest-with-django/blob/master/requirements-dev.txt) we've can add in the two packages to be installed.
+
+```
+pytest==2.7.2
+pytest-django==2.8.0
+```
+
+> These were the latest versions when this post was written, be sure to check pypi if either [`pytest`](https://pypi.python.org/pypi/pytest) or [`pytest-django`](https://pypi.python.org/pypi/pytest-django) have been updated.
+
+We need to bootstrap `pytest` with our Django project [settings](https://docs.djangoproject.com/en/1.8/topics/settings/).
+
+`pytest-django`'s documentation [recommends](https://pytest-django.readthedocs.org/en/latest/configuring_django.html) a few different ways to achieve this. I'd advocate loading them in through a global `conftest.py`
+
+In `pytest` any `conftest.py` files are invoked before any tests are run. They provide a convenient method to setup hooks or configure any settings for our tests.
+Where the `conftest.py` lives dictates the scope of where it applies. If present in the root test folder, hooks declared will apply to all tests. If present in a specific module, hooks will only apply to tests in that module.
+
+```
+tests/
+├── conftest.py # Applies to all tests
+└── example/
+    ├── conftest.py # Applies only to tests in this module/folder.
+    ├── test_foo.py
+```
+
+In our case, we want the [`conftest.py`](https://github.com/TrackMaven/using-pytest-with-django/blob/master/tests/conftest.py) to apply to all tests therefore we place it in our tests root directory.
+
+```
+import os
+import django
+from django.conf import settings
+
+# We manually designate which settings we will be using in an environment variable
+# This is similar to what occurs in the `manage.py`
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'app.config.settings')
+
+
+# `pytest` automatically calls this function once when tests are run.
+def pytest_configure():
+    settings.DEBUG = False
+    # If you have any test specific settings, you can declare them here,
```

Comment:
Do you think removing the commented-out lines here would help clarify what's actually going on in this function?